test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3531 @@
+          expiry,
+          isPut
+        );
+
+        const optionTerms = [
+          underlying,
+          strike,
+          underlying,
+          expiry,
+          strikePrice,
+          this.isPut ? 2 : 1,
+          params.strikeAsset,
+        ];
+
+        await expect(
+          this.vault.connect(managerSigner).commitAndClose(optionTerms)
+        ).to.be.revertedWith(this.isPut ? "!put" : "!call");
+      });
+
+      it("reverts when the expiry is before the delay", async function () {
+        const block = await provider.getBlock();
+
+        const underlying = params.asset;
+        const strike = params.strikeAsset;
+        const collateral = params.collateralAsset;
+        const strikePrice = parseEther(params.firstOptionStrike.toString());
+        const isPut = params.isPut;
+
+        let expiryDate;
+
+        const currentBlock = moment.utc(block.timestamp * 1000);
+
+        // get the same day's 8am
+        const sameDay8AM = moment
+          .utc()
+          .year(currentBlock.year())
+          .month(currentBlock.month())
+          .date(currentBlock.date())
+          .hour(8)
+          .minute(0)
+          .second(0)
+          .millisecond(0);
+
+        if (currentBlock.isBefore(sameDay8AM)) {
+          expiryDate = sameDay8AM;
+        } else {
+          // use next day's 8am
+          expiryDate = sameDay8AM.add(1, "day");
+        }
+        const expiry = Math.round(expiryDate.valueOf() / 1000);
+
+        await this.oTokenFactory.createOtoken(
+          underlying,
+          strike,
+          collateral,
+          strikePrice.div(BigNumber.from("10").pow(BigNumber.from("10"))),
+          expiry,
+          isPut
+        );
+        await time.increaseTo(
+          expiryDate.subtract(OPTION_DELAY, "seconds").valueOf() / 1000
+        );
+
+        const optionTerms = [
+          underlying,
+          strike,
+          collateral,
+          expiry,
+          strikePrice,
+          this.optionType,
+          params.strikeAsset,
+        ];
+
+        await expect(
+          this.vault.connect(managerSigner).commitAndClose(optionTerms)
+        ).to.be.revertedWith("Option expiry cannot be before delay");
+      });
+
+      it("sets the next option and closes existing short", async function () {
+        const res = await this.vault
+          .connect(managerSigner)
+          .commitAndClose(this.optionTerms, { from: manager });
+
+        const receipt = await res.wait();
+        const block = await provider.getBlock(receipt.blockNumber);
+
+        assert.equal(await this.vault.nextOption(), this.oTokenAddress);
+        assert.equal(
+          (await this.vault.nextOptionReadyAt()).toNumber(),
+          parseInt(block.timestamp) + OPTION_DELAY
+        );
+        assert.isTrue((await this.vault.lockedAmount()).isZero());
+        assert.equal(await this.vault.currentOption(), constants.AddressZero);
```

Comment:
Need to check that the yvWETH is returned after `commitAndClose`

Reply:
👍